diff --git a/title.field.inc b/title.field.inc
index d94905b..b87e9f8 100644
--- a/title.field.inc
+++ b/title.field.inc
@@ -97,7 +97,7 @@ function title_field_formatter_settings_summary($field, $instance, $view_mode) {
  */
 function title_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, $items, $display) {
   $settings = $display['settings'];
-  $output = isset($items[0]) ? $items[0]['safe_value'] : '';
+  $output = isset($items[0]) ? $items[0]['value'] : '';
 
   if (!empty($output) && $settings['title_link'] == 'content') {
     $uri = entity_uri($entity_type, $entity);
