diff --git a/metatag.metatag.inc b/metatag.metatag.inc
index f88271a..94c0417 100644
--- a/metatag.metatag.inc
+++ b/metatag.metatag.inc
@@ -166,6 +166,15 @@ function metatag_metatag_info() {
     ),
   );
 
+  // When the maxlength module is enabled, add a limit to the description and
+  // abstract metatag.
+  if (module_exists('maxlength')) {
+    $info['tags']['description']['form']['#maxlength_js_limit'] = 150;
+    $info['tags']['description']['form']['#maxlength_js'] = TRUE;
+    $info['tags']['abstract']['form']['#maxlength_js_limit'] = 150;
+    $info['tags']['abstract']['form']['#maxlength_js'] = TRUE;
+  }
+
   $info['tags']['keywords'] = array(
     'label' => t('Keywords'),
     'description' => t("A comma-separated list of keywords about the page. This meta tag is <em>not</em> supported by most search engines."),
