diff --git a/fieldable_panels_panes.module b/fieldable_panels_panes.module
index 1b98ac7..d475ea3 100644
--- a/fieldable_panels_panes.module
+++ b/fieldable_panels_panes.module
@@ -41,6 +41,7 @@ function fieldable_panels_panes_entity_info() {
       'id' => 'fpid',
       'revision' => 'vid',
       'bundle' => 'bundle',
+      'label' => 'title', // This key is required for proper integration with Title module.
       'language' => 'language',
       'uuid' => 'uuid',
       'revision uuid' => 'vuuid',
@@ -198,7 +199,12 @@ function fieldable_panels_panes_field_extra_fields_display_alter(&$displays, $co
   if ($context['entity_type'] == 'fieldable_panels_pane' && $context['view_mode'] == 'full') {
     // Hide display of the title field on the 'full' view mode (because it gets
     // displayed as the Pane title).
-    $displays['title']['visible'] = FALSE;
+
+    // Ensure the 'title' key exists, since some other modules may have remove it. This
+    // is the case of the Title module for example.
+    if (isset($displays['title'])) {
+      $displays['title']['visible'] = FALSE;
+    }
   }
 }
 
@@ -762,6 +768,15 @@ function fieldable_panels_panes_preprocess_panels_pane(&$vars) {
   }
 }
 
+/**
+ * Process variables for fieldable-panels-pane.tpl.php.
+ */
+function fieldable_panels_panes_process_fieldable_panels_pane(&$variables) {
+  if (module_exists('title')) {
+    title_field_replacement_hide_label('fieldable_panels_pane', $variables['elements']['#element'], $variables['content']);
+  }
+}
+
 // -------------------------------------------------------------------------
 // Database and general entity API functions
 
